internal/storage/writer/writer.go

```diff
@@ -68,9 +68,16 @@ func ForCompaction(config *config.Compaction, monitor monitor.Monitor, store sto
 		}
 	}
 
+	// Crate the flusher
 	monitor.Info("server: setting up compaction %T to run every %.0fs...", writer, interval.Seconds())
-	flusher := flush.New(monitor, writer, nameFunc)
-	return compact.New(store, flusher, flusher, monitor, interval)
+
+	// TODO: once we have everything working, consider making the flusher per writer (requires changing all writers)
```

Comment:
Should we also think of batching commits and making the batch size configurable?

Reply:
Good point, maybe in a separate PR we could do that. Right now we're already batching (compaction) and this batching is currently time-based (configurable).